.claude/scripts/release_testing/lib/git_ops.py

```
@@ -0,0 +1,172 @@
+            all commits on the branch.
+
+    Returns:
+        list[dict]: Each dict has keys: sha, subject, date.
+    """
+    tag = find_version_tag(since_version) if since_version else None
+    if tag:
+        revision = f"{tag}..origin/{branch}"
+    else:
+        revision = f"origin/{branch}"
+
+    result = subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)